### PR TITLE
[RDF] Add the `Stats` method to RInterface to get TStatistic instances

### PR DESCRIPTION
**Pull request by dpiparo**
out of columns of a dataset.
The implementation is trivial and relies on the already present `RInterface::Fill`.
The `TStatistic::Merge` implementation needed to be changed to be well behaved in a MT context.
Other changes are present in the PR to allow better handling of the filling of non-TH1 objects via the `RInterface::Fill` method.

### CODE REVIEW

**Comment by dpiparo** [comment]
Adding @gganis as author of `TStatistics`.

**Comment by dpiparo** [comment]
@phsft-bot build

File: tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx [hunk 1]
```diff
@@ -245,6 +245,14 @@ extern template void FillHelper::Exec(unsigned int, const std::vector<int> &, co
...
 FillHelper::Exec(unsigned int, const std::vector<unsigned int> &, const std::vector<unsigned int> &);
 
+template<typename T>
+void SetDirZero(T &){}
+
+inline void SetDirZero(TH1 &h)
+{
+   h.SetDirectory(nullptr);
+}
+
```
Comment (bluehood) [inline tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx]: The template version `SetDirZero(T &)` will always have precedence over `SetDirZero(TH1 &h)` unless you really pass a reference to `TH1` (not a reference to `TH1D`, for example). I don't think this is the desired behavior..?

(also I'd suggest `UnsetDirectory` as an alternative function name)

File: math/mathcore/src/TStatistic.cxx [hunk 2]
```diff
@@ -92,25 +92,48 @@ void TStatistic::Print(Option_t *) const {
...
+
+   // Merge the statistic quantities into local variables to then 
+   // update the data members of this object
+   auto firstStatPtr = statPtrs[0];
+   auto N = firstStatPtr->fN;
+   auto M = firstStatPtr->fM;
+   auto M2 = firstStatPtr->fM2;
+   auto W = firstStatPtr->fW;
+   auto W2 = firstStatPtr->fW2;
+   for (auto i = 1U; i < nStatsPtrs; ++i) {
```
Comment (bluehood) [inline math/mathcore/src/TStatistic.cxx]: it's a bit surprising that you put `this` at position 0 in `statPtrs` only to then special-case it and loop only from 1 onwards.
maybe `this` should not be put in `statPtrs` at all?

File: tree/dataframe/inc/ROOT/RDF/InterfaceUtils.hxx [hunk 3]
```diff
@@ -135,7 +122,7 @@ struct HistoUtils {
 };
 
 template <typename T>
-struct HistoUtils<T, true> {
+struct HistoUtils<T, false> {
    static void SetCanExtendAllAxes(T &) {}
    static bool HasAxisLimits(T &) { return true; }
```
Comment (bluehood) [inline tree/dataframe/inc/ROOT/RDF/InterfaceUtils.hxx]: I guess it's not true that anything that is not a ROOT6 TH1 has axes limits

File: tree/dataframe/inc/ROOT/RDF/RInterface.hxx [hunk 4]
```diff
@@ -1372,6 +1373,34 @@ public:
...
+      }
+      const auto validColumnNames = GetValidatedColumnNames(1, columns);
+      return Fill(TStatistic(), validColumnNames);
+   }
+
+   ////////////////////////////////////////////////////////////////////////////
+   /// \brief Return a TStatistic object, filled once per event (*lazy action*)
+   ///
+   /// \param[in] Column with the values to fill the statistics with.
+   /// \param[in] Column with the weights to fill the statistics with.
```
Comment (bluehood) [inline tree/dataframe/inc/ROOT/RDF/RInterface.hxx]: missing argument names: `\param[in] value`, `\param[in] weight`

File: math/mathcore/src/TStatistic.cxx (hunk 2 above)
Comment (dpiparo) [reply math/mathcore/src/TStatistic.cxx]: It is not special. One needs to merge the TStatistics objects which have at least one entry.

File: tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx (hunk 1 above)
Comment (dpiparo) [reply tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx]: Thanks for spotting this.

**Pull request merged**